### DISCUSSION TITLE
RELATIONS.md - access relation Id

### DISCUSSION THREAD

**Discussion by BenOnTrack** [systemed/tilemaker#749] [Q&A] [answered]
The current documentation **RELATIONS.md** states : _You can obtain the relation ID and role from `rel`, which is a list (a two-element Lua table). The first element (`rel[1]`) is the id, the second (`rel[2]`) the role. Remember Lua tables are 1-indexed!_

But I've realized that under the **node_function** (I haven't tried with the **way_function**), I actually access the relation Id via `rel` (instead of `rel[1]` since rel seems to be a number). 

if you do:

```
while true do
		local rel = NextRelation()
		if not rel then break end
		print(rel)
end
```

you will get the relation Id.

Am I doing anything wrong ? Or is there really a typo in the doc ?

**Comment by cldellow** [comment, marked as answer]
I think you're right -- the docs are incorrect. `NextRelation()` returns two return values, not a single return value that is an array with two values.

The docs might read better as:

---

You can also obtain the element's role in the relation at the same time, via a second return value:

```lua
local rel, role = NextRelation()
if not rel then break end
```

**Reply by BenOnTrack** [reply]
That sounds right! Thanks for confirming.
